File: model/song.go
package model

import (
	"time"

	"github.com/allape/gocrud"
)

type Song struct {
	gocrud.Base
	Name        string `json:"name"`
	Subtitle    string `json:"subtitle"`
	Filename    string `json:"filename"`
	Cover       string `json:"cover"`
	Digest      string `json:"digest"`
	MIME        string `json:"mime"`
	FFProbeInfo string `json:"ffprobeInfo"`
	Description string `json:"description"`
}

type SongLyrics struct {
	SongID    gocrud.ID `json:"songId" gorm:"primaryKey"`
	LyricsID  gocrud.ID `json:"lyricsId" gorm:"primaryKey"`
	CreatedAt time.Time `json:"createdAt" gorm:"autoCreateTime;<-:create"`
}

File: ui/admin/src/model/song.ts
import { IBase, IBaseSearchParams } from "@allape/gocrud";
import { ICollection } from "./collection.ts";
import { ILyrics } from "./lyrics.ts";

export interface ISong extends IBase {
  name: string;
  subtitle: string;
  filename: string;
  cover: string;
  digest: string;
  mime: string;
  ffprobeInfo: string;
  description: string;
}

export interface ISongSearchParams extends IBaseSearchParams {
  keywords?: string;
  like_name?: string;
  like_collectionName?: string;
  in_collectionId?: ICollection["id"][];
}

export interface ISongLyrics extends Pick<IBase, "createdAt"> {
  songId: ISong["id"];
  lyricsId: ILyrics["id"];
}
